Tests/DatadogTests/Datadog/DatadogCore/DatadogCoreTests.swift

```diff
@@ -0,0 +1,190 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+@testable import Datadog
+
+private struct FeatureMock: DatadogFeature {
+    struct Event: Encodable {
+        let event: String
+    }
+
+    var name: String
+    var requestBuilder: FeatureRequestBuilder = FeatureRequestBuilderMock()
+    var messageReceiver: FeatureMessageReceiver = FeatureMessageReceiverMock()
+}
+
+class DatadogCoreTests: XCTestCase {
+    override func setUp() {
+        super.setUp()
+        temporaryCoreDirectory.create()
+    }
+
+    override func tearDown() {
+        temporaryCoreDirectory.delete()
+        super.tearDown()
+    }
+
+    func testWhenWritingEventsWithDifferentTrackingConsent_itOnlyUploadsAuthorizedEvents() throws {
+        let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
```

Comment:
I took the approach of testing `DatadogCore` "end-to-end". We're using the [existing `ServerMock`](https://github.com/DataDog/dd-sdk-ios/blob/919ebef125b4103033400dc84bf53de5db0dd349/Tests/DatadogTests/Datadog/Mocks/ServerMock.swift#L19) here - it intercepts `URLSession` (with `URLProtocol`). We collect events in mocked V2 Feature and make an assertion on events passed by core for building upload request.

Because it is touching no internals of the `DatadogCore`, such tests should be well sustainable and require no maintenance during regular code changes.